internal/resources/gitrepository/resource_git_repository.go

```diff
@@ -17,26 +29,309 @@ import (
+		statusKey:            status.StatusSchema,
+	}
+
+	innerMap := map[string]*schema.Schema{
+		spec.SpecKey: spec.SpecSchema,
+	}
+
+	for key, value := range innerMap {
+		if isDataSource {
+			gitRepositorySchema[key] = helper.UpdateDataSourceSchema(value)
+		} else {
+			gitRepositorySchema[key] = value
+		}
+	}
+
+	return gitRepositorySchema
+}
+
+func resourceGitRepositoryCreate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	gitRepositoryName, ok := d.Get(nameKey).(string)
+	if !ok {
+		return diag.Errorf("unable to read git repository name")
+	}
+
+	gitRepositoryNamespaceName, ok := d.Get(namespaceNameKey).(string)
+	if !ok {
+		return diag.Errorf("unable to read git repository namespace name")
+	}
+
+	scopedFullnameData := scope.ConstructScope(d, gitRepositoryName, gitRepositoryNamespaceName)
+
+	if scopedFullnameData == nil {
+		return diag.Errorf("Unable to create Tanzu Mission Control git repository entry; Scope full name is empty")
+	}
+
+	var (
+		UID  string
+		meta = common.ConstructMeta(d)
+	)
+
+	err := enableContinuousDelivery(&config, scopedFullnameData, meta)
+	if err != nil {
+		return diag.FromErr(errors.Wrapf(err, "Unable to create Tanzu Mission Control git repository entry, name : %s", gitRepositoryName))
+	}
+
+	switch scopedFullnameData.Scope {
+	case commonscope.ClusterScope:
+		if scopedFullnameData.FullnameCluster != nil {
+			gitRepositoryReq := &gitrepositoryclustermodel.VmwareTanzuManageV1alpha1ClusterNamespaceFluxcdGitrepositoryGitRepositoryRequest{
+				GitRepository: &gitrepositoryclustermodel.VmwareTanzuManageV1alpha1ClusterNamespaceFluxcdGitrepositoryGitRepository{
+					FullName: scopedFullnameData.FullnameCluster,
+					Meta:     meta,
+					Spec:     spec.ConstructSpecForClusterScope(d),
+				},
+			}
+
+			gitRepositoryResponse, err := config.TMCConnection.ClusterGitRepositoryResourceService.VmwareTanzuManageV1alpha1ClusterFluxcdGitrepositoryResourceServiceCreate(gitRepositoryReq)
+			if err != nil {
+				return diag.FromErr(errors.Wrapf(err, "Unable to create Tanzu Mission Control cluster git repository entry, name : %s", gitRepositoryName))
+			}
+
+			UID = gitRepositoryResponse.GitRepository.Meta.UID
+		}
+	case commonscope.ClusterGroupScope:
+		if scopedFullnameData.FullnameClusterGroup != nil {
+			gitRepositoryReq := &gitrepositoryclustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupNamespaceFluxcdGitrepositoryGitRepositoryRequest{
+				GitRepository: &gitrepositoryclustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupNamespaceFluxcdGitrepositoryGitRepository{
+					FullName: scopedFullnameData.FullnameClusterGroup,
+					Meta:     meta,
+					Spec:     spec.ConstructSpecForClusterGroupScope(d),
+				},
+			}
+
+			gitRepositoryResponse, err := config.TMCConnection.ClusterGroupGitRepositoryResourceService.VmwareTanzuManageV1alpha1ClustergroupFluxcdGitrepositoryResourceServiceCreate(gitRepositoryReq)
```

Comment:
nit : `VmwareTanzuManageV1alpha1ClustergroupFluxcdGitrepositoryResourceServiceCreate` this is too long. 
I believe going forward this can be cutshort and VmwareTanzuManage is not needed

Reply:
Sure

Reply:
Can be taken up as a refactoring task later. Since it is out of scope for the changes related to the MR, I think a separate MR would be better which does this across all the models and clients.